### PR TITLE
Try to make README more approachable.

### PR DESCRIPTION
**Pull request by mahesh-hegde**
* Explain more basic concepts up-front before going into example code.
  - TODO: We should probably skip implementation details in shown bindings.

* Add an FAQ section, describing how to resolve common errors.

* Fix some small typos etc..

* There's some informal language (gracious usage of `you`, guess it's hard to write READMEs like RFCs).

closes: #210

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,10 +4,26 @@
...
+
+* _Outputs_: Output can be generated in package-structured (one file per class) or single file bindings. Target path to write C and Dart bindings needs to be specified.
+
+* _Classes_: Specify which classes or packages you need bindings for. Specifying a package includes all classes inside it recursively.
+
+Check out the [examples](jnigen/example/) to see some sample configurations.
+
+C code is always generated into a directory with it's own build configuration. It's built as a separate dynamic library.
+
+Lastly, [dart_only bindings](#pure-dart-bindings) mode is also available as a proof-of-concept, which does not need intermediate C bindings apart from dependency on the support library `package:jni`.
```
Comment (dcharkes) [inline README.md:23]:
```suggestion
Lastly, [dart_only bindings](#pure-dart-bindings) mode is also available as a proof-of-concept. It does not need intermediate C bindings, only a dependency on the support library `package:jni`.
```

File: README.md [hunk 2]
```diff
@@ -4,10 +4,26 @@
...
 
 ## Example
-It's possible to generate bindings for libraries, or any Java source files.
+It's possible to generate bindings for libraries, or Java source files.
```
Comment (dcharkes) [inline README.md:26]: libraries? Did you mean JAR files?

File: README.md [hunk 3]
```diff
@@ -168,15 +164,32 @@ For example, on Powershell:
...
+#### `jnigen` is not finding classes.
+Ensure you are providing correct source and class paths, and they follow standard directory structure. If your class name is `com.abc.MyClass`, `MyClass` must be in `com/abc/MyClass.java` relative to one of the source paths, or `com/abc/MyClass.class` relative to one of the class paths specified in YAML.
+
+If the classes are in JAR file, make sure to provide path to JAR file itself, and not the containing directory.
+
+#### `jnigen` is unable to parse sources.
+If the errors are similar to `symbol not found`, ensure all dependencies of the source are available. If such dependency is compiled, it can be included in `class_path`.
+
+#### Should I use `jnigen` over Method channels?
+This is currently an experimental package. Many features are missing, and it's rough around the edges. You're welcome to try it and give feedback.
```
Comment (dcharkes) [inline README.md:192]: @HosseinYousefi What is the preferred wording for where we are with this product?

File: README.md [hunk 4]
```diff
@@ -256,3 +269,6 @@ However there are 2 caveats to this caveat.
 
 The JAR files (`$SDK_ROOT/platforms/android-$VERSION/android.jar`) can be used instead. But compiled JARs do not include JavaDoc and method parameter names. This JAR is automatically included by Gradle when `android_sdk_config` >> `add_gradle_deps` is specified.
 
+## Contributing
+See the wiki for architecture-related documents.
```
Comment (dcharkes) [inline README.md:273]: We should add a CONTRIBUTING.md (can do in a separate PR.)

For inspiration: https://github.com/dart-lang/ffigen/blob/master/CONTRIBUTING.md

The markdown file can link to the wiki.

**Review by dcharkes — APPROVED** [review]
lgtm! I like it!

Please also let @HosseinYousefi take a look.

File: README.md (hunk 3 above)
Comment (HosseinYousefi) [reply README.md:192]: This looks good to me. @mit-mit What do you think?

**Comment by HosseinYousefi** [comment]
LGTM! Waiting for an answer on `#### Should I use `jnigen` over Method channels?` comment.

**Comment by HosseinYousefi** [comment]
I think it's good for now. We can change it later. Merging...

**Pull request merged**